crates/services/p2p/src/service.rs

```diff
@@ -412,6 +413,11 @@ pub struct Task<P, V, B, T> {
     heartbeat_max_time_since_last: Duration,
     next_check_time: Instant,
     heartbeat_peer_reputation_config: HeartbeatPeerReputationConfig,
+    // cached view
+    cached_view: Arc<CachedView>,
+    // milliseconds wait time between cache reset
+    cache_reset_interval: Duration,
+    next_cache_reset_time: Instant,
```

Comment:
Looks like it could be an internal logic of the `CachedView` and on each insert/get we can clean it.

Reply:
it could be, but I wanted to not make the `get_from_cache_or_db` require a mutable reference to `Self` because it's just a getter. no strong opinion here, so if you want it that way, i can move it around